### PR TITLE
Add KakuteF7 and Kopis 2

### PR DESCRIPTION
**Pull request by bkueng**
- Missing: video
- Needs fixing: the BL and Firmware updating is duplicated from Omnibus (it works the same way except for a different bootloader file), we can move it out into a common page (@hamishwillee , do you have a good idea where?).

### CODE REVIEW

File: en/flight_controller/kakutef7.md [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+* microSD (for logging)
+* 6 UARTs
+* 1 I2C bus
+* 6 PWM outputs
+* built-in OSD chip (AB7456 via SPI)
+
+
+## Where to Buy
+
+The board can be bought for example from one of the following shops:
```
Comment (hamishwillee) [inline en/flight_controller/kakutef7.md]: > The board can be bought from one of the following shops (for example):

or

> The board can be bought, for example, from one of the following shops:

File: en/flight_controller/kakutef7.md [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+  * RAM: 320 KB SRAM
+  * FLASH: 1 MB
+* Standard racer form factor: 36x36 mm with standard 30.5 mm hole pattern
+* ICM20689 Accel / Gyro (Soft-mounted)
+* BMP280 Baro
+* microSD (for logging)
+* 6 UARTs
+* 1 I2C bus
+* 6 PWM outputs
+* built-in OSD chip (AB7456 via SPI)
```
Comment (hamishwillee) [inline en/flight_controller/kakutef7.md]: Built-in

File: en/flight_controller/kakutef7.md [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+
+
+## Where to Buy
+
+The board can be bought for example from one of the following shops:
+- [Holybro](https://shop.holybro.com/kakute-f7_p1104.html) 
+- [getfpv](https://www.getfpv.com/holybro-kakute-f7-tekko32-f3-metal-65a-4-in-1-esc-combo.html)
+
+> **Tip** The Kakute F7 fits with the Tekko32 4-in-1 ESC.
```
Comment (hamishwillee) [inline en/flight_controller/kakutef7.md]: By "fits" do you mean "is designed to work with?" or "comes with"? If the former, might be worth saying that.

Not a general rule, but often it is easier to parse product names visually if they are in italic. So you might if you can be bothered markup everywhere product as *Kakute F7*.

File: en/flight_controller/kakutef7.md [hunk 4]
```diff
@@ -0,0 +1,158 @@
...
+| R2, T2 | UART2 RX and TX | TELEM2 |
+| R3, T3 | UART3 RX and TX | NuttX debug console |
+| R4, T4 | UART4 RX and TX | GPS1 |
+| R6, T6 | UART6 RX and TX | RC port |
+| R7, T7 | UART7 RX and TX (RX is located in the plug for use with 4-in-1 ESCs) ||
+| LED | WS2182 addressable LED signal wire (not tested) ||
+| Buz- | Piezo buzzer negative leg (Connect buzzer positive leg to 5v pad) ||
+| 3V3 | 3.3v output (200 mA max) ||
+| M1 to M4 | Motor signal outputs (located in plug) ||
+| M5, M6 | Additional motor signal outputs (located at side of board) ||
```
Comment (hamishwillee) [inline en/flight_controller/kakutef7.md]: located "on" side of board

File: en/flight_controller/kakutef7.md [hunk 5]
```diff
@@ -0,0 +1,158 @@
...
+| G or GND | Ground ||
+| SDA, SCL | I2C connection (for peripherals) ||
+| R1, T1 | UART1 RX and TX | TELEM1 |
+| R2, T2 | UART2 RX and TX | TELEM2 |
+| R3, T3 | UART3 RX and TX | NuttX debug console |
+| R4, T4 | UART4 RX and TX | GPS1 |
+| R6, T6 | UART6 RX and TX | RC port |
+| R7, T7 | UART7 RX and TX (RX is located in the plug for use with 4-in-1 ESCs) ||
+| LED | WS2182 addressable LED signal wire (not tested) ||
+| Buz- | Piezo buzzer negative leg (Connect buzzer positive leg to 5v pad) ||
```
Comment (hamishwillee) [inline en/flight_controller/kakutef7.md]: 5V

File: en/flight_controller/kakutef7.md [hunk 6]
```diff
@@ -0,0 +1,158 @@
...
+| 3V3 | 3.3v output (200 mA max) ||
+| M1 to M4 | Motor signal outputs (located in plug) ||
```
Comment (hamishwillee) [inline en/flight_controller/kakutef7.md]: This is "plug for use with 4-in-1 ESCs)". Might be worth adding in full.

File: en/flight_controller/kakutef7.md [hunk 7]
```diff
@@ -0,0 +1,158 @@
...
+git clone --recursive  https://github.com/PX4/Bootloader.git
+cd Bootloader
+make omnibusf4sd_bl
+```
+
+#### Flash Bootloader
+
+You can flash the PX4 bootloader using the [dfu-util](http://dfu-util.sourceforge.net/) or the graphical [dfuse](https://www.st.com/en/development-tools/stsw-stm32080.html) tool on windows.
+
+Don't be afraid to try flashing using any of the below methods.
```
Comment (hamishwillee) [inline en/flight_controller/kakutef7.md]: ... any of the methods below (the STM32 MCU cannot be bricked).

File: en/flight_controller/kakutef7.md [hunk 8]
```diff
@@ -0,0 +1,158 @@
...
+
+```
+dfu-util -a 0 --dfuse-address 0x08000000 -D  build/omnibusf4sd_bl/omnibusf4sd_bl.bin
+```
+
+Reboot the flight controller and it let it boot without holding the boot button.
+
+##### dfuse
+
+See the dfuse manual is here: https://www.st.com/resource/en/user_manual/cd00155676.pdf
```
Comment (hamishwillee) [inline en/flight_controller/kakutef7.md]: remove "is"

File: en/flight_controller/kakutef7.md [hunk 9]
```diff
@@ -0,0 +1,158 @@
...
+
+Flash the `omnibusf4sd_bl.bin` file.
```
Comment (hamishwillee) [inline en/flight_controller/kakutef7.md]: FWIW I tend to use bold for file names, code for terminal commands. This is fine if you have been consistent across doc though.

File: en/flight_controller/omnibus_f4_sd.md [hunk 10]
```diff
@@ -25,11 +25,11 @@ These are the main differences compared to a [Pixracer](../flight_controller/pix
 * Standard racer form factor: 36x36 mm with standard 30.5 mm hole pattern
 * MPU6000 Accel / Gyro
 * BMP280 Baro (not all boards have it mounted)
-* microSD (logging and parameters)
+* microSD (for logging)
 * Futaba S.BUS and S.BUS2 / Spektrum DSM2 and DSMX / Graupner SUMD / PPM input / Yuneec ST24
 * OneShot PWM out (configurable)
 * built-in current sensor
-* built-in OSD chip (AB7456 via SPI, not supported yet)
+* built-in OSD chip (AB7456 via SPI)
```
Comment (hamishwillee) [inline en/flight_controller/omnibus_f4_sd.md]: Built-in

File: en/flight_controller/omnibus_f4_sd.md [hunk 11]
```diff
@@ -25,11 +25,11 @@ These are the main differences compared to a [Pixracer](../flight_controller/pix
 * Standard racer form factor: 36x36 mm with standard 30.5 mm hole pattern
 * MPU6000 Accel / Gyro
 * BMP280 Baro (not all boards have it mounted)
-* microSD (logging and parameters)
+* microSD (for logging)
 * Futaba S.BUS and S.BUS2 / Spektrum DSM2 and DSMX / Graupner SUMD / PPM input / Yuneec ST24
 * OneShot PWM out (configurable)
 * built-in current sensor
```
Comment (hamishwillee) [inline en/flight_controller/omnibus_f4_sd.md]: Built-in

**Comment by hamishwillee** [comment]
@bkueng Looks great. Regarding firmware update and replacing betaflight....

I'd create new doc as a peer of this one http://docs.px4.io/master/en/advanced_config/bootloader_update.html . Both docs will need a clear differentiating name. In this case probably **PX Bootloader Update onto Betaflight Systems**

**Comment by bkueng** [comment]
Thanks @hamishwillee. I created a new Betaflight BL update page and I think I addressed all comments.
The video is commented out for now.

File: en/advanced_config/bootloader_update_from_betaflight.md [hunk 12]
```diff
@@ -0,0 +1,67 @@
...
+
+### Bootloader Update using Source
+
+#### Download Bootloader Source
+
+Download and build the [Bootloader](https://github.com/PX4/Bootloader) via:
+```
+git clone --recursive  https://github.com/PX4/Bootloader.git
+cd Bootloader
+make <target> # For example: omnibusf4sd_bl
```
Comment (hamishwillee) [inline en/advanced_config/bootloader_update_from_betaflight.md]: Do you think we could have the option for KakuteF7 here too?

File: en/advanced_config/bootloader_update_from_betaflight.md [hunk 13]
```diff
@@ -0,0 +1,67 @@
+# Bootloader Flashing onto Betaflight Systems
+
+This page documents how to flash the PX4 bootloader onto boards preflashed with Betaflight.
```
Comment (hamishwillee) [inline en/advanced_config/bootloader_update_from_betaflight.md]: I think worth having an "e.g. " with links to the two boards we know about.

File: en/advanced_config/bootloader_update_from_betaflight.md [hunk 14]
```diff
@@ -0,0 +1,67 @@
...
+##### Enter DFU mode
+
+Both methods require the board to be in DFU mode. 
+To enter DFU mode, hold the boot button down while connecting the USB cable to your computer. 
+The button can be released after the board is powered up.
+
+##### dfu-util
+
+```
+dfu-util -a 0 --dfuse-address 0x08000000 -D  build/omnibusf4sd_bl/omnibusf4sd_bl.bin
```
Comment (hamishwillee) [inline en/advanced_config/bootloader_update_from_betaflight.md]: As above, good to make this "target" and then explain what the path is, or keep the path and explain that they should look in similar place if installing similar binaries.

File: en/advanced_config/bootloader_update_from_betaflight.md [hunk 15]
```diff
@@ -0,0 +1,67 @@
...
+```
+
+Reboot the flight controller and it let it boot without holding the boot button.
+
+##### dfuse
+
+See the dfuse manual here: https://www.st.com/resource/en/user_manual/cd00155676.pdf
+
+Flash the **omnibusf4sd_bl.bin** file.
```
Comment (hamishwillee) [inline en/advanced_config/bootloader_update_from_betaflight.md]: Ditto. Good to generalise

**Comment by hamishwillee** [comment]
@bkueng Looks good. I would generalise the Bootloader doc a bit more - it refers to omnibusf4sd a bit too often. But if you disagree with the need, then perhaps just both examples in the introduction to that doc.

**Comment by hamishwillee** [comment]
And when you're happy with that, good to merge.

**Comment by bkueng** [comment]
> I would generalise the Bootloader doc a bit more - it refers to omnibusf4sd a bit too often. But if you disagree with the need, then perhaps just both examples in the introduction to that doc.

I was wondering about that, and since you mention it as well, I did the change.

**Comment by hamishwillee** [comment]
Looks great. Thanks!

**Pull request merged**